pandas/io/excel.py

```diff
@@ -465,7 +465,7 @@ def _parse_cell(cell_contents, cell_typ):
                 if not squeeze or isinstance(output[asheetname], DataFrame):
                     output[asheetname].columns = output[
                         asheetname].columns.set_names(header_names)
-            except StopIteration:
+            except EmptyDataError:
```

Comment:
prove it by having tests that raise these separtely (this may not exist or partially), that is the point, want to prove that we are only making very deliberate API changes.

IOW, pretend you are a user reading the release notes, I want to know what changes I can expect. So the whatsnew should show exactly what is changing.